joblib/parallel.py

```diff
@@ -915,7 +965,9 @@ def __call__(self, iterable):
             # The main thread will consume the first pre_dispatch items and
             # the remaining items will later be lazily dispatched by async
             # callbacks upon task completions.
-            iterator = itertools.islice(iterator, pre_dispatch)
+
+            # TODO: this iterator should be batch_size * n_jobs
+            iterator = itertools.islice(iterator, self._pre_dispatch_amount)
```

Comment:
This TODO means that you would like to have the default `pre_dispatch` to `'auto'` and send `2*n_jobs*batch_size` in `pre_dispatch_amount`?

Reply:
Yes, although it could be in a different fix: in the default case, `batch_size = "auto"` meaning that at this point in the code, `batch_size = 1`. So `2*n_jobs*batch_size=2*n_jobs`, and what you suggest is already what happens.

In the case where `batch_size` is set to an actual number, `pre_dispatch` should indeed to me (and to you :)) have a default value of `batch_size*n_jobs*2` and not `2*n_jobs`. But this issue predates this PR. Perhaps it deserves another PR + news entry, as this will lead to a change in the default value.

Reply:
I agree on the separate PR :)